LiftLog.Ui/Models/SessionHistoryDao/SessionHistoryDaoV2.cs

```diff
@@ -111,14 +111,29 @@ public static PotentialSetDaoV2 FromModel(Lib.Models.PotentialSet model) =>
 
 internal partial class RecordedSetDaoV2
 {
-    public RecordedSetDaoV2(int repsCompleted, TimeOnlyDao completionTime)
+    public RecordedSetDaoV2(int repsCompleted, DateOnlyDao? completionDate, TimeOnlyDao? completionTime)
     {
         RepsCompleted = repsCompleted;
+        CompletionDate = completionDate;
         CompletionTime = completionTime;
     }
 
-    public static RecordedSetDaoV2? FromModel(Lib.Models.RecordedSet? model) =>
-        model is null ? null : new(model.RepsCompleted, model.CompletionTime);
+    public static RecordedSetDaoV2? FromModel(Lib.Models.RecordedSet? model)
+    {
+        if (model is null)
+            return null;
+            
+        DateOnlyDao? nonNullDate = model.CompletionDate;
+        TimeOnlyDao? nonNullTime = model.CompletionTime;
+        
+        return new(model.RepsCompleted, nonNullDate, nonNullTime);
+    }
 
-    public Lib.Models.RecordedSet ToModel() => new(RepsCompleted, CompletionTime);
+    public Lib.Models.RecordedSet ToModel()
+    {
+        DateOnlyDao nonNullDate = CompletionDate ?? new DateOnlyDao(DateOnly.FromDateTime(DateTime.Today));
+        TimeOnlyDao nonNullTime = CompletionTime ?? new TimeOnlyDao(TimeOnly.FromDateTime(DateTime.Now));
+        
+        return new Lib.Models.RecordedSet(RepsCompleted, nonNullDate, nonNullTime);
+    }
```

Comment:
Unfortunately this won't be backwards compatible and when it loads a save it will set all completed set dates to today.

What we'll need to do is pass the `DateOnly sessionDate` as a parameter to this function, and just pass it down the chain

Reply:
Actually so there's no need to make the time nullable either